crates/iceberg-catalog/src/service/tabular_idents.rs

```diff
@@ -0,0 +1,81 @@
+use iceberg::TableIdent;
+use std::fmt::{Display, Formatter};
+use std::ops::Deref;
+use uuid::Uuid;
+
+#[derive(Hash, PartialOrd, PartialEq, Debug, Clone, Copy, Eq)]
+pub enum TabularIdentUuid {
+    Table(Uuid),
+    View(Uuid),
+}
+
+impl TabularIdentUuid {
+    #[must_use]
+    pub fn typ_str(&self) -> &'static str {
+        match self {
+            TabularIdentUuid::Table(_) => "table",
+            TabularIdentUuid::View(_) => "view",
+        }
+    }
+}
+
+impl Display for TabularIdentUuid {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", &**self)
+    }
+}
+
+// We get these two types since we are using them as HashMap keys. Those need to be sized,
+// implementing these types via Cow makes them not sized, so we go for two... not ideal.
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub(crate) enum TabularIdentRef<'a> {
+    // TODO: TableIdent is from iceberg-rust, AFAIK, TableIdent and ViewIdent are the same, should we
```

Comment:
I would keep it the way it is for now - maybe iceberg-rust switches to tabular in the future as well.

I stumbled over the Name "Ref" at first, as many Refs (SchemaRef) etc are Arcs. I don't have a better idea though.

Reply:
Can rename it to Borrowed, I guess that's more in line with Cow's nomenclature

Reply:
renamed to Borrowed